qiskit/transpiler/runningpassmanager.py

```diff
@@ -107,11 +108,40 @@ def _to_target(self, passmanager_ir: DAGCircuit, in_program: QuantumCircuit) ->
         circuit = dag_to_circuit(passmanager_ir, copy_operations=False)
         circuit.name = self.metadata["output_name"]
 
-        if self.property_set["layout"] is not None:
+        layout = self.property_set["layout"]
+        # If  there is no initial layout set but "original_layout" is set that means
+        # `ElidePermutations` ran and we have a final layout. Since a final layout
+        # needs an initial layout to be useful we take the original layout field which
+        # is just a trivial layout at the time ElidePermutations ran (which must be
+        # before an initial layout gets set).
+        if not layout:
+            layout = self.property_set["original_layout"]
+        final_layout = self.property_set["final_layout"]
```

Comment:
Same comment here, do we need to bother setting `final_layout` if `layout` is `None`?

Reply:
I think yes, although this begs the questions why you would like to set a final layout without an initial layout. :-)